### PR TITLE
add support to Bundle easyblock to install list of components

### PR DESCRIPTION
**Pull request by boegel**
better approach than ~~#979~~ to handle easyconfigs like https://github.com/hpcugent/easybuild-easyconfigs/pull/3440

### CODE REVIEW

**Comment by boegel** [comment]
example usage:

```
== temporary log file in case of crash /tmp/eb-OMLiSq/easybuild-72WK6v.log
== processing EasyBuild easyconfig /example/X11-20160819-intel-2016b.eb
== building and installing X11/20160819-intel-2016b...
== fetching files...
== creating build dir, resetting environment...
== unpacking...
== patching...
== preparing...
== configuring...
== building...
== testing...
== installing...
== installing bundle component xorg-macros v1.19.0 (1/32)...
== installing bundle component fixesproto v5.0 (2/32)...
== installing bundle component glproto v1.4.17 (3/32)...
== installing bundle component xextproto v7.3.0 (4/32)...
== installing bundle component xcb-proto v1.11 (5/32)...
== installing bundle component inputproto v2.3.1 (6/32)...
== installing bundle component xproto v7.0.28 (7/32)...
== installing bundle component libpthread-stubs v0.3 (8/32)...
== installing bundle component kbproto v1.0.7 (9/32)...
== installing bundle component xtrans v1.3.5 (10/32)...
== installing bundle component xineramaproto v1.2.1 (11/32)...
== installing bundle component randrproto v1.5.0 (12/32)...
== installing bundle component renderproto v0.11 (13/32)...
== installing bundle component libXau v1.0.8 (14/32)...
== installing bundle component libxcb v1.11.1 (15/32)...
== installing bundle component xcb-util v0.4.0 (16/32)...
== installing bundle component xcb-util-image v0.4.0 (17/32)...
== installing bundle component xcb-util-keysyms v0.4.0 (18/32)...
...
```

@wpoely86, @hajgato, @ocaisa: thoughts?

File: easybuild/easyblocks/generic/bundle.py [hunk 1]
```diff
@@ -56,6 +62,43 @@ def __init__(self, *args, **kwargs):
...
+            cfg = self.cfg.copy()
+
+            cfg['name'] = comp_name
+            cfg['version'] = comp_version
+            cfg.generate_template_values()
+
+            # do not inherit easyblock to use from parent (since that would result in an infinite loop in install_step)
+            cfg['easyblock'] = None
+
+            for key in comp_specs:
```
Comment (wpoely86) [inline easybuild/easyblocks/generic/bundle.py]: so `comp_specs` is again a tuple?

File: easybuild/easyblocks/generic/bundle.py [hunk 2]
```diff
@@ -71,8 +114,44 @@ def build_step(self):
...
+            # run relevant steps
+            comp.patch_step()
+            comp.configure_step()
+            comp.build_step()
+            comp.install_step()
+
+            # update environment to ensure stuff provided by former components can be picked up by latter components
+            # once the installation is finalised, this is handled by the generated module
+            reqs = comp.make_module_req_guess()
+            for envvar in reqs:
```
Comment (wpoely86) [inline easybuild/easyblocks/generic/bundle.py]: does this give any  logging output?

File: easybuild/easyblocks/generic/bundle.py (hunk 1 above)
Comment (boegel) [inline easybuild/easyblocks/generic/bundle.py]: no, a dict (key-value)

File: easybuild/easyblocks/generic/bundle.py (hunk 2 above)
Comment (boegel) [inline easybuild/easyblocks/generic/bundle.py]: `env.setvar` does the logging

**Comment by wpoely86** [comment]
```
== 2016-08-23 18:02:08,657 environment.py:97 INFO Environment variable MANPATH set to /apps/gent/CO7/x86-64/software/X11/20160819-intel-2016b/man:/apps/gent/CO7/x86-64/software/X11/20160819-intel-2016b/share/man:/apps/gent/CO7/x86-64/software/X11/20160819-intel-2016b/man:/apps/gent/CO7/x86-64/software/X11/20160819-intel-2016b/share/man:/apps/gent/CO7/x86-64/software/X11/20160819-intel-2016b/man:/apps/gent/CO7/x86-64/software/X11/20160819-intel-2016b/share/man:/apps/gent/CO7/x86-64/software/X11/20160819-intel-2016b/man:/apps/gent/CO7/x86-64/software/X11/20160819-intel-2016b/share/man:/apps/gent/CO7/x86-64/software/X11/20160819-intel-2016b/man:/apps/gent/CO7/x86-64/software/X11/20160819-intel-2016b/share/man:/apps/gent/CO7/x86-64/software/X11/20160819-intel-2016b/man
```
Do you notice the pattern :wink:

**Comment by boegel** [comment]
@wpoely86 nice catch, should be fixed now, thanks for the review!

**Comment by boegel** [comment]
fix for remark verified, tested with https://github.com/hpcugent/easybuild-easyconfigs/pull/3440, good to go, thanks for the review @wpoely86!

**Pull request merged**